pkgs/applications/science/programming/fdr/default.nix

```diff
@@ -0,0 +1,87 @@
+{stdenv, fetchurl, qt57, ncurses, xorg, zlib, python27Packages}:
+let libs = [
+  stdenv.cc.cc
+  python27Packages.python
+  qt57.qtbase
+  qt57.qtx11extras
+  ncurses
+  xorg.libX11
+  xorg.libXft
+  zlib
+  ];
+in
+stdenv.mkDerivation {
+  name = "fdr";
+  src = fetchurl {
+    url = https://www.cs.ox.ac.uk/projects/fdr/downloads/fdr-3754-linux-x86_64.tar.gz;
+    md5 = "c09a05a5d9b54fc65ed12de8b7cdfc11";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+  installPhase = ''
+    mkdir -p "$out"
+
+    # shipped Qt is buggy
+    rm lib/libQt*
+    rm -r lib/qt_plugins
+
+    cp -r * "$out"
+    # Hack around lack of libtinfo in NixOS
+    ln -s ${ncurses.out}/lib/libncursesw.so.6 $out/lib/libtinfo.so.5
+    ln -s ${qt57.qtbase.out}/lib/qt5/plugins $out/lib/qt_plugins
+    ln -s ${zlib.out}/lib/libz.so.1 $out/lib/libz.so.1
+
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" "$out/bin/fdr4"
```

Comment:
If I do as you suggest, I get exactly this in the output:

```
$ patchelf --print-rpath fdr4
/nix/store/gs36amglhgii6i9nb7wrlps2lnvcq4f6-gcc-5.4.0-lib/lib:/nix/store/a21ahiimxzb3gc9vgdvxqlv4p4n5iph8-python-2.7.13/lib:/nix/store/36zs803khc4p017nai2xpvnim35jcq2h-qtbase-5.7.0/lib:/nix/store/000433zcc8rsks02n8w9gdjgak8xcra3-qtx11extras-5.7.0/lib:/nix/store/v6swva6d3hdj334r7n9ddl3c23snd9f1-ncurses-6.0/lib:/nix/store/80zzabd2yrz9ickskv3zz5z672h6mrv6-libX11-1.6.4/lib:/nix/store/bqkammrvcm5bn073mzs0h9c7wqyddbs2-libXft-2.3.2/lib:/nix/store/6rhawr786hmpg0lw910rna03v0gbw93j-zlib-1.2.8/lib:$out/lib
```

It doesn't seem like $out is expanded.